dapr/proto/runtime/v1/conversation.proto

```diff
@@ -0,0 +1,342 @@
+  optional double temperature = 7;
+}
+
+// ConversationRequestV1Alpha2 is the new request object for Conversation.
+// Many of these fields are inspired by openai.ChatCompletionNewParams
+// https://github.com/openai/openai-go/blob/main/chatcompletion.go#L2106
+message ConversationRequestV1Alpha2 {
+  // The name of Conversation component
+  string name = 1;
+
+  // The ID of an existing chat (like in ChatGPT)
+  optional string context_id = 2;
+
+  // Inputs for the conversation, support multiple input in one time.
+  // This is the revamped conversation inputs better matching openai.
+  repeated ConversationInputV1Alpha2 inputs = 3;
+
+  // Parameters for all custom fields.
+  map<string, google.protobuf.Any> parameters = 4;
+
+  // The metadata passing to conversation components.
+  map<string, string> metadata = 5;
+
+  // Scrub PII data that comes back from the LLM
+  optional bool scrub_pii = 6;
+
+  // Temperature for the LLM to optimize for creativity or predictability
+  optional double temperature = 7;
+
+  // Tools register the tools available to be used by the LLM during the conversation.
+  // These are sent on a per request basis.
+  // The tools available during the first round of the conversation
+  // may be different than tools specified later on.
+  repeated ConversationTools tools = 8;
+
+  // Controls which (if any) tool is called by the model. 
+  // `none` means the model will not call any tool and instead generates a message. 
+  // `auto` means the model can pick between generating a message or calling one or more tools. 
+  // `required` means the model must call one or more tools. 
+  // Specifying a particular tool via:
+  // `{"type": "function", "function": {"name": "my_function"}}` forces the model to call that tool.
+  // `none` is the default when no tools are present.
+  // `auto` is the default if tools are present.
+  // ref: https://github.com/openai/openai-go/blob/main/chatcompletion.go#L1976
+  // ref: https://python.langchain.com/docs/how_to/tool_choice/
+  optional google.protobuf.StringValue tool_choice = 9;
+
+  /* TODO: fields to bring in from openai.ChatCompletionNewParams after Dapr 1.16.
+  Note: They can be within the other messages, not necessarily top level here.
+  MaxCompletionTokens: https://github.com/openai/openai-go/blob/main/chatcompletion.go#L2131
+  Audio: https://github.com/openai/openai-go/blob/main/chatcompletion.go#L2186
+  Modalities: https://github.com/openai/openai-go/blob/main/chatcompletion.go#L2215
+  ReasoningEffort: https://github.com/openai/openai-go/blob/main/chatcompletion.go#L2224
+  Stop: https://github.com/openai/openai-go/blob/main/chatcompletion.go#L2249 <----------------should we bring this in?
+  StreamOptions: https://github.com/openai/openai-go/blob/main/chatcompletion.go#L2251
+  ResponseFormat: https://github.com/openai/openai-go/blob/main/chatcompletion.go#L2284
+  */
+}
+
+// maintained for backwards compatibility
+message ConversationInput {
```

Comment:
Please, version these proto fields with the relevant version suffix.

Reply:
this is the old proto... so I did not do this to break existing clients

Reply:
Could you perhaps schedule the change for post 1.16?

Reply:
> Could you perhaps schedule the change for post 1.16?
As is make the original ConversationInput become ConversationInputV1?

Reply:
I recognize that this is not reflected in the PR conversation; however, this was a direct ask from Josh in a chat I had with him. He's on vacation rn. I can move it back no problem, but I will create an issue to separate the protos out eventually.